crates/scroll/node/src/builder/engine.rs

```diff
@@ -60,20 +63,55 @@ where
     fn validate_version_specific_fields(
         &self,
         _version: EngineApiMessageVersion,
-        _payload_or_attrs: PayloadOrAttributes<'_, Self::ExecutionData, ScrollPayloadAttributes>,
+        payload_or_attrs: PayloadOrAttributes<'_, Self::ExecutionData, ScrollPayloadAttributes>,
     ) -> Result<(), EngineObjectValidationError> {
+        validate_scroll_payload_or_attributes(
+            &payload_or_attrs,
+            payload_or_attrs.message_validation_kind(),
+        )?;
         Ok(())
     }
 
     fn ensure_well_formed_attributes(
         &self,
         _version: EngineApiMessageVersion,
-        _attributes: &ScrollPayloadAttributes,
+        attributes: &ScrollPayloadAttributes,
     ) -> Result<(), EngineObjectValidationError> {
+        validate_scroll_payload_or_attributes(
+            &PayloadOrAttributes::PayloadAttributes::<'_, ExecutionData, _>(attributes),
+            MessageValidationKind::PayloadAttributes,
+        )?;
+
+        // ensure block data hint is present pre euclid.
```

Comment:
this means that we need to prioritize the fetching of the pre euclid block data hints.